### PR TITLE
Add control flow api: case

### PR DESCRIPTION
**Pull request by liym27**
- ### Add a control flow API: `case`
```
case(pred_fn_pairs, default=None, name=None)
```
> #### `case` works like an `if-elif-elif-else` chain.
- ### API document
- Chinese Docment: https://github.com/PaddlePaddle/FluidDoc/pull/1629

- English docment:
image

### CODE REVIEW

File: python/paddle/fluid/layers/control_flow.py [hunk 1]
```diff
@@ -1797,6 +1797,131 @@ def cond(pred, true_fn=None, false_fn=None, name=None):
     return merged_output
 
 
+def _error_message(what, arg_name, op_name, right_value, error_value):
```
Comment (zhhsplendid) [inline python/paddle/fluid/layers/control_flow.py:1801]: I saw you raise TypeError in _error_message and also raise at where you use _error_message, that's not good.

Two options:

Don't raise TypeError in _error_message, but return the error message as a string in _error_message

Remove "raise TypeError" here, rename "_error_message" to raise_error_with_message or some other name properly.

Comment (liym27) [reply python/paddle/fluid/layers/control_flow.py:1801]: Thanks. Done according to option 1.

**Review by zhhsplendid — DISMISSED** [review]
LGTM now

**Review by heavengate — DISMISSED** [review]
LG API

**Review by lanxianghit — DISMISSED** [review]

**Review by lanxianghit — APPROVED** [review]

**Review by heavengate — APPROVED** [review]

**Pull request merged**